### PR TITLE
LLT-5223  ignore connection closed error on pyro daemon shutdown

### PR DESCRIPTION
**Pull request by Jauler**
### Problem
There seems to be a race when shutting down libtelio RPC services started for testing purposes. The race nature is that shutdown is requested via RPC command, which races between connection being closed and response being sent to the client. This PR allows for any order of events to be accepted.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: nat-lab/tests/telio.py [hunk 1]
```diff
@@ -657,7 +657,7 @@ async def on_stderr(stdout: str) -> None:
                     if self._process.is_executing():
                         await self.stop_device()
                         self._quit = True
-                    self.get_proxy().shutdown()
+                    self.get_proxy().shutdown(self._connection.target_name())
```
Comment (mathiaspeters) [inline nat-lab/tests/telio.py:660]: It would have been enough to pass `self._node.name` but this works too :smile:

Comment (Jauler) [reply nat-lab/tests/telio.py:660]: oh, I missed it :D

Comment (Jauler) [reply nat-lab/tests/telio.py:660]: oh wait, but is'nt `self._node.name` just normally `alpha` or `beta` or similar? In that case, maybe it is a bit better to have specific container name?

Comment (mathiaspeters) [reply nat-lab/tests/telio.py:660]: I think the most important thing is to be able to see if the shutdown is happening multiple times on the same client, which the node name would tell us, but your method would make it easier to perhaps see patterns over time so let's go with it

**Comment by ThrasherLT** [comment]
lgtm, +1.0

**Review by mathiaspeters — APPROVED** [review]
Looks good! +2

**Pull request merged**